collector/receiver.go

```diff
@@ -0,0 +1,41 @@
+package collector
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+)
+
+type TxReceiver interface {
+	SendTx(ctx context.Context, tx *TxIn) error
+}
+
+type HTTPReceiver struct {
+	url string
+}
+
+func NewHTTPReceiver(url string) *HTTPReceiver {
+	return &HTTPReceiver{
+		url: url,
+	}
+}
+
+func (r *HTTPReceiver) SendTx(ctx context.Context, tx *TxIn) error {
+	rawTx, err := tx.Tx.MarshalBinary()
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, r.url, bytes.NewReader(rawTx))
```

Comment:
isn't a http request per transaction overkill? i'd imagine it's better to use a TCP connection and send only rlp tx separated by newline